test/middleware/spec.back-link.js

```diff
@@ -1,11 +1,20 @@
-var backLinks = require('../../lib/middleware/back-links');
-
-var util = require('util'),
+var proxyquire = require('proxyquire'),
+    util = require('util'),
     EventEmitter = require('events').EventEmitter;
 
+var getBackLinks = function (previousSteps) {
+    return proxyquire('../../lib/middleware/back-links', {
+        '../util/helpers': {
+            getPreviousSteps: function () {
+                return previousSteps;
+            }
+        }
+    });
+};
```

Comment:
We are unable to use sinon to stub the getPreviousSteps method as it isn't a part of the back-link middleware but an external dependency. In order to force the outcome of getPreviousSteps we need to proxyquire it every time with different return values